src/codegen/llvm/codegen_llvm_visitor.cpp

```diff
@@ -895,6 +898,256 @@ void CodegenLLVMVisitor::visit_program(const ast::Program& node) {
     }
 
     logger->debug("Dumping generated IR...\n" + dump_module());
+    // Setup CodegenHelper for C++ wrapper file
+    setup(node);
+    print_wrapper_routines();
+    print_target_file();
+}
+
+void CodegenLLVMVisitor::print_wrapper_headers_include() {
+    print_standard_includes();
+    print_coreneuron_includes();
+}
+
+void CodegenLLVMVisitor::print_mechanism_range_var_structure() {
+    printer->add_newline(2);
+    printer->add_line("/** Instance Struct passed as argument to LLVM IR kernels */");
+    printer->start_block("struct {} "_format(mod_filename + instance_struct_type_name));
+    for (const auto& variable: instance_var_helper.instance->get_codegen_vars()) {
+        auto is_pointer = variable->get_is_pointer();
+        auto name = to_nmodl(variable->get_name());
+        auto qualifier = is_constant_variable(name) ? k_const() : "";
```

Comment:
I've actually been thinking whether this is needed for the llvm backend since the members of the InstanceStruct on the LLVM IR are not constant if I am not mistaken

Reply:
I am not sure if it makes any difference...may be we can ask this next week to George if this is useful/harmful.

Reply:
As discussed with George and Pramod `const` qualifier in the members of the struct for LLVM IR don't make a difference. However since this is a wrapper script in the C++ level we can still keep these